### ISSUE TITLE
Metamask unusable after adding token with 0.1 decimals?

### ISSUE THREAD

**Issue by Leozolt** [MetaMask/metamask-extension#12032]
Hello.

Using Metamask extension 10.0.3 on Firefox
Binance Smart Chain
Ubuntu 20.04

got into serious trouble after adding a "token" / bep-721 NFT with "0.1" decimals.

Metamask doesn't load anymore. The only thing that I can do is Unlock and after that I get the error message:
Message: div() number type has more than 15 significant digits: 1.2589254117941673

Code: BigNumber Error

Pleas advise **how to change that data** of decimals for the token to 1 or more. It might be a more serious issue but the first thing I want to try is to change the value fro 0.1 to 1. But I do not know where it is stored.
![Screenshot from 2021-09-07 22-40-15](https://user-images.githubusercontent.com/26518970/132410946-2653bc05-9eb6-409f-a049-621271e72acb.png)

If 0.1 should not be possible to enter and save, please debug?

I also tried to interact with the contract of AIRT NFT's via bscscan because I tried to send the token/NFT to another adress. Now I can't use Metamask. I still think it's the 0.1 value for "Decimals" that buggs..

Screenshot shows data for AIRT token that I added.

Best Regards,

Tristan

**Comment by Leozolt** [comment]
I removed Metamask extension an installed it again to recover functionallity.

I also tried to replicate the error but did not succeed on the first try.

I added the BSC token with 0.1 decimals on adress "A".
--I was able to transfer the token from adress A to adress B successfully via BSC contract interaction on bscscan (safeTransferForm)
--I was able to approve adress A to spend the token from adress B with adress B credetials of course.

And there it is again...less severily (not fatal) but still occurring. If I switch from adress B to adress A:     Message: div() number type has more than 15 significant digits: 1.2589254117941673

    Code: BigNumber Error


If I switch from another adress to adress A there is no error.

Weird...

The contract is 0xf5db804101d8600c26598a1ba465166c33cdaa4b and my token is tokenId 136129 .. but I am not sure how it would only give that errors on specific tokens/contract combinations?

Is it the 0.1 decimals that bugs the metamask?
